pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -129,19 +146,33 @@ func (t *TopologyCache) AddHints(si *SliceInfo) ([]*discovery.EndpointSlice, []*
 		allocatedHintsByZone[zone] += diff
 	}
 
+	if len(allocatedHintsByZone) == 0 {
+		klog.V(2).Infof("No hints allocated for zones, removing them from %s EndpointSlices for %s Service", si.AddressType, si.ServiceKey)
+		// TODO(liorlieberman)
+		t.RemoveHints(si.ServiceKey, si.AddressType)
+		slicesToCreate, slicesToUpdate = RemoveHintsFromSlices(si)
+		return slicesToCreate, slicesToUpdate, events
+
+	}
```

Comment:
@robscott please verify it is the correct behavior here (I move it from SetHints and we did not call  `RemoveHintsFromSlices` there) 

+ 
Should we publish an event here, I assume yes but not sure since it wasnt on the DD.

Reply:
Yep, I think this is a reasonable move as long as it doesn't require significant unit test refactoring. It looks like SetHints() is only called from this function. I agree that publishing an event is a good idea here.

Reply:
why we do not call `RemoveHintsFromSlices` here?

Reply:
We did not call `RemoveHintsFromSlices` in this case previously. I added it. Is that the correct behavior?
see [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/endpointslice/topologycache/topologycache.go#L139-L143)